### PR TITLE
Code refactor

### PR DESCRIPTION
**Pull request by tezc**
A bit of code refactor, to list a few things : 

- Added `raft_node_set_next_idx()` to the API which can be used by the application after loading the snapshot.
- Removed duplicate declarations for functions in raft_private.h and raft.h
- Removed leftover code pieces
- Removed unnecessary malloc casts.
- Fixed some Clang-tidy warnings:
   - Fixed type conversations 
   - Removed const declarations in function prototypes for basic data types.
   - Removed unused imports
   - Renamed symbols starting with "__". It is reserved by the C implementation.

### CODE REVIEW

File: src/raft_server.c [hunk 1]
```diff
@@ -143,6 +139,185 @@ void raft_clear(raft_server_t* me_)
...
+        case RAFT_LOGTYPE_ADD_NODE:
+            raft_node_set_voting(node, 0);
+            break;
+
+        default:
+            assert(0);
+            break;
+    }
+}
+
```
Comment (tezc) [inline src/raft_server.c:320]: Shows a giant diff but it's mostly unchanged, just moved up in the file.

File: src/raft_log.c [hunk 2]
```diff
@@ -185,14 +183,14 @@ raft_entry_t** log_get_from_idx(log_t* me_, raft_index_t idx, int *n_etys)
...
-    int logs_till_end_of_log;
+    long logs_till_end_of_log;
 
     if (i < me->back)
         logs_till_end_of_log = me->back - i;
     else
         logs_till_end_of_log = me->size - i;
 
-    *n_etys = logs_till_end_of_log;
+    *n_etys = (int) logs_till_end_of_log;
```
Comment (yossigo) [inline src/raft_log.c:193]: Did you consider changing the prototype to `long` as well?

**Review by yossigo — DISMISSED** [review]

File: src/raft_log.c (hunk 2 above)
Comment (tezc) [reply src/raft_log.c]: Do you suggest propagating this to upper layers? (e.g through `get_batch()` callback) 

Probably, we won't ever want a giant appendreq message. A better way, split it and send multiple appendreqs to followers so a follower can flush it/process it while the next one is on the way). 

I've changed the prototype of this function but added `int` cast to caller of this function (__log_get_batch) which does the boundary checks anyway. I believe it's simpler keep `int` for upper layers. Log implementations can handle this, they call multiple times if they want more entries. What do you think?

**Review by yossigo — APPROVED** [review]

File: src/raft_log.c (hunk 2 above)
Comment (yossigo) [reply src/raft_log.c]: @tezc Missed this comment. I agree we don't want a huge batch but I'm not sure if it's the responsibility of this function to handle that.

**Pull request merged**